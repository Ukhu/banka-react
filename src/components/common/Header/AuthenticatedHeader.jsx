import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const AuthenticatedHeader = ({ user, link }) => (
  <header>
    <div className="container">
      <div id="header-title">
        <Link to={`/${link}`}>
          <h1>Banka</h1>
        </Link>
      </div>
      <div id="header-links">
        <p>
          <i className="far fa-user" />
          {' '}
          {user}
        </p>
        <Link
          id="logout"
          to="/"
          onClick={() => {
            window.localStorage.removeItem('user');
            window.localStorage.removeItem('accountDetails');
          }}
        >
        Logout
        </Link>
      </div>
    </div>
  </header>
);

AuthenticatedHeader.propTypes = {
  user: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
};

export default AuthenticatedHeader;
