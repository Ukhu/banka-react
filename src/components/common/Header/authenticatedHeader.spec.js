import React from 'react';
import { configure, shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import AuthenticatedHeader from './AuthenticatedHeader';

configure({ adapter: new Adapter() });

describe('Authenticated Header component <Authenticated />', () => {
  const user = 'John';
  const link = 'dashboard';

  it('renders correctly', () => {
    const wrapper = shallow(
      <AuthenticatedHeader user={user} link={link} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('has the header tag', () => {
    const wrapper = shallow(
      <AuthenticatedHeader user={user} link={link} />,
    );

    expect(wrapper.find('header')).toHaveLength(1);
  });

  it('has 2 links', () => {
    const wrapper = shallow(
      <AuthenticatedHeader user={user} link={link} />,
    );

    expect(wrapper.find('Link')).toHaveLength(2);
  });
});
