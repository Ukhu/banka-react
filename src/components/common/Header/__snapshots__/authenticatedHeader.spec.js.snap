// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticated Header component <Authenticated /> renders correctly 1`] = `ShallowWrapper {}`;
